database-memory-agent/tools.py

```
@@ -0,0 +1,39 @@
+    ]
+
+    results = vector_collection.aggregate(pipeline)
+    array_of_results = []
+    for doc in results:
+        array_of_results.append(doc)
+    return array_of_results
+
+
+def calculator_tool(user_input: str) -> str:
+    try:
+        result = eval(user_input)
```

Ruff (0.14.8)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)